lib/features/signup/ui/widgets/signup_button.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:mentorship/core/helpers/extensions.dart';
+import 'package:mentorship/core/widgets/loading_dialog.dart';
+import 'package:mentorship/core/widgets/toast.dart';
+import 'package:mentorship/features/signup/logic/cubits/signup_cubit.dart';
+import 'package:mentorship/features/signup/logic/cubits/signup_state.dart';
+import '../../../../core/widgets/app_main_button.dart';
+
+class SignupButton extends StatelessWidget {
+  const SignupButton({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocListener<SignupCubit, SignupState>(
+      listenWhen: (previous, current) =>
+          current is SignupLoading ||
+          current is SignupSuccess ||
+          current is SignupFail,
+      listener: (context, state) {
+        state.whenOrNull(
```

Comment:
I mean like this @themohabgamal 

this is comment for mohab don't change anything @NourhanHamada